### DISCUSSION TITLE
A huge disappointment

### DISCUSSION THREAD

**Discussion by Citrus0299** [ghostfolio/ghostfolio#3666] [General]
I have been playing around with the self hosted version of the app for a few days now and I would like to share my experiences.

## TLDR
Very good concept, privacy focussed and a really good idea, but deeply flawed in every day use. Missing error handling in every corner, no proper documentation and an underperforming tech stack. It's overall not fun to use and sometimes painful in very basic tasks.

## Initial Setup
The setup instructions are ok and easy to follow, but after that you are completely on your own. No documentation whatsoever.
Figuring out why the hell historical currency aren't being imported took too way too long. Searching github issues before finding the deeply hidden button to manually fetch historical data.
Also the app crashed after a few minutes because redis cannot save snapshots to disk as there is no persistant volume configured for it in the provided docker-compose.

## UI/UX
The web ui is very modern and seems really nice at first, until you try to actually use it.
So I tried to recreate my trading portfolio starting out with an initial cash balance manually.
To this day I have no idea what the button "Update Cash Balance" when creating a buy or interest action actually does. No tooltip, no helptext, nothing. It also doesn't work for any historical data. So I already had to give up on my workflow and had to completely disregard the cash balance and fix it later. Interest is also super weirdly calculated, it seemingly will be added to your portfolio like a stock which you can't sell. I would expect that it would be added to the cash balance instead.
Error handling is very much only existent on forms, everything else will fail silently or with a completely unhelpful error message.
Checking the container logs for everything is really annoying, but a really good way to hide the underlying flaws.

## Performance
The performance and resource consumption is really really bad. I don't have any words on how a webserver can consume 700 MB of RAM while doing absolutely nothing. Oh my god it's JS on the server combined with postgres for time series data, it's destined to be slow. I don't know how you are able to sustain a business with the these hosting costs, but good job. Maybe this does explain the pricing.

**Comment by dtslvr** [comment]
Hi @Citrus0299

Thank you for sharing your feedback, I really appreciate it. I’m sorry to hear that some aspects of the app have been disappointing.

Based on your feedback, I think these are the key areas to focus:

* Documentation: Creating a getting started guide
* UI/UX: Refining error handling
* Performance: Optimizing memory usage

Your input would help improve the project. If you’re interested in contributing to any of these tasks, please feel free to reach out.

Best regards,
Thomas

**Comment by Dikootje** [comment]
I share a lot of @Citrus0299 feelings. 

I've been playing around for a couple of hours over the past few days with both the free and the self-hosted options. 

Documentation needs more than just a getting started guide. There's lots of options in the interface and no description on how to use these except for searching the discussion on github where you might find a comment. 

The import is one of the first features I tried to get a feel how it would reflect on 1 of my accounts. However the csv import is a huge let down in it's current state. 
- The currency field does not accept EUR
- Empty/null values in Fee are not accepted (Why not simply use 0 if there's no fee provided)
- The fields should be exactly as specified. It would be much more friendly if one could simply upload a csv and the point to which columns hold which information. 

So I've played around with some dummy activities. It looks really nice and most of the times the UI responds very well. I hope this project gets some traction and matures as an FOSS solution compared to all the overpriced commercial options already out there.

**Comment by karter16** [comment]
There needs to be a fourth category to this IMHO, and that that is financial accuracy. There are a number of issues that have been languishing in the issues list for months that directly affect the core financial accuracy of the product. I find it astonishing that these haven't been prioritised to be resolved, as I don't understand the point of working on anything else on the product if the core information it's presenting is not correct.

#3330 
#3492 
#3320 
#3416 
#3329 
#3233 
#2895 
#2700

**Reply by cladand** [reply]
You can add #4341
I agree with you

**Comment by Citrus0299** [comment]
Hi @dtslvr,

thank you for addressing my feedback. 
I have been asking around on several OSS discord servers and the response is very much the same. Ghostfolio in its current state isn't really useable. I didn't go into much detail in my initial post, but there are a lot of things that need to be addressed.

### General
- [ ] Proper documentation for general use and self-hosting setup
- [ ] Clarify how interest works
- [ ] Postgres performance tuning for time series data and memory optimizations

### UI/UX
- [ ] Error handling with useful error messages
- [ ] Tooltips for checkboxes, buttons, ...
- [ ] Allow to manually calculate the exchange rate when adding a BUY order in any currency.

### CSV Import, thanks to @Dikootje 
- [x] The currency field does not accept EUR
- [ ] Empty/null values in Fee are not accepted (Why not simply use 0 if there's no fee provided)
- [ ] The fields should be exactly as specified. It would be much more friendly if one could simply upload a csv and the point to which columns hold which information.

### Financial Accuracy thanks to @karter16 
The math should be 100% correct if the user supplied accurate data. If there are small inaccuracies, it should be pointed out how significant they are why they exist. One example could be currency conversion rates, which get fetched on a daily basis, but are actually much more granular. 
- [ ] #3330
- [ ] #3492
- [ ] #3320
- [ ] #3416
- [ ] #3329
- [ ] #3233
- [ ] #2895
- [ ] #2700

### Admin UI
The Market Data and job queue are in a very rough state, but functional.
- [ ] Redesign all dropdowns, it's super hard to navigate
- [ ] The job queue doesn't show any completed tasks

#### The reason I gave up and decided to write this
After fixing the provided docker-compose file, figuring out how to use the app and struggling to add any stock, because I haven't fetched the historical data yet. I was adding one order after another and was actually pretty pleased with the experience. Until I added a somewhat special ticker: DKDF.F. To this day I have no idea why the hell historical data can't be fetched on my self-hosted instance for this stock, there are no errors, nothing. BUT it does work in the cloud version of the app, which shouldn't be possible, since it is the exact same application.

### Update 11.09.2024
After seeing a month full of rapid development I decided to revisit this app again.
I was able to find almost all of the missing stock data by manually querying yahoo finance and copying the ISIN. I added these to a new account. After filtering between the three accounts individually, the whole filtering mechanism broke. I am now completely unable to filter by an individual account even after restarting. The only error message available is in the console is this:
```js
ERROR TypeError: Cannot destructure property 'Symbol(Symbol.iterator)' of '(0 , A.groupBy)(...)' as it is undefined.
    at D.buildFiltersAsQueryParams (main.e8988407ffc72116.js:3:144561)
    at D.fetchPortfolioPerformance (main.e8988407ffc72116.js:3:150241)
    at e.update (2999.2984cc92123714c4.js:1:46732)
    at Object.next (2999.2984cc92123714c4.js:1:45226)
    at Ee.next (main.e8988407ffc72116.js:3:767318)
    at ye._next (main.e8988407ffc72116.js:3:766875)
    at ye.next (main.e8988407ffc72116.js:3:766453)
    at r._subscribe (main.e8988407ffc72116.js:3:755820)
    at D._trySubscribe (main.e8988407ffc72116.js:3:758651)
    at M._trySubscribe (main.e8988407ffc72116.js:3:764032)
```

**Comment by fib1985** [comment]
Hi to all,

Despite the app is really interesting, something must be addressed, for example I agree with all these points (especially the accredit "interest" one):

"To this day I have no idea what the button "Update Cash Balance" when creating a buy or interest action actually does. No tooltip, no helptext, nothing. It also doesn't work for any historical data. So I already had to give up on my workflow and had to completely disregard the cash balance and fix it later. **Interest is also super weirdly calculated, it seemingly will be added to your portfolio like a stock which you can't sell. I would expect that it would be added to the cash balance instead.**
Error handling is very much only existent on forms, everything else will fail silently or with a completely unhelpful error message."
